federationsender/queue/queue.go

```diff
@@ -66,6 +72,7 @@ func (oqs *OutgoingQueues) getQueue(destination gomatrixserverlib.ServerName) *d
 			incomingPDUs:    make(chan *gomatrixserverlib.HeaderedEvent, 128),
 			incomingEDUs:    make(chan *gomatrixserverlib.EDU, 128),
 			incomingInvites: make(chan *gomatrixserverlib.InviteV2Request, 128),
+			retryServerCh:   make(chan bool),
```

Comment:
We don't seem to `close` this anywhere - I suspect the rate of leakage is low, but still not ideal. If we have a way to check if the worker is really blocked (as per the other comment) then we can `defer` a closure of this at the beginning of `backgroundSend()`.

Reply:
We don't close any of the other channels so I don't see a problem here.

Reply:
You make a good point.